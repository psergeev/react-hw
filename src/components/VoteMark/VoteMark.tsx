import * as React from 'react';
import './VoteMark.scss';

interface Props {
    vote: number;
}

export default React.memo((props: Props) => (
    <div className="vote-mark">{props.vote}</div>
));
