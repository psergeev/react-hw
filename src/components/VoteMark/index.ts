import VoteMark from './VoteMark';

export default VoteMark;
