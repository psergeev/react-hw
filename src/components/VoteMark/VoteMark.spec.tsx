import * as React from 'react';
import * as renderer from 'react-test-renderer';
import VoteMark from '.';

test('VoteMark component works as expected', () => {
    const component = renderer.create(
        <VoteMark vote={7.6} />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
