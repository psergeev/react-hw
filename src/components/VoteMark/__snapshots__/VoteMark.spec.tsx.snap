// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoteMark component works as expected 1`] = `
<div
  className="vote-mark"
>
  7.6
</div>
`;
